src/handlers/Marriage.cpp

```diff
@@ -43,21 +43,30 @@ void Marriage_Open(Character *character, PacketReader &reader)
 	}
 }
 
+enum MarriageRequestType : unsigned char
+{
+	MarriageApproval = 0,
+	Divorce = 1,
+};
+
 // Requesting marriage approval
 void Marriage_Request(Character *character, PacketReader &reader)
 {
-	unsigned char request_type = reader.GetChar();
-	int session_id = reader.GetInt();
+	MarriageRequestType request_type = static_cast<MarriageRequestType>(reader.GetChar());
+	/*int session_id = */reader.GetInt();
 	reader.GetByte();
 	std::string name = reader.GetEndString();
 
-	if (character->npc_type == ENF::Law)
+	if (character->npc_type != ENF::Law)
+		return;
+
+	switch (request_type)
 	{
-		if (request_type == 1) // Marriage approval
+		case MarriageApproval:
```

Comment:
Looks like `MarriageApproval` should be 1 and `Divorce` should be 0 in the enum

Reply:
Fixed.